### PR TITLE
Replace nosetests with pytest

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**

### CODE REVIEW

File: src/__init__.py [hunk 1]
```diff
@@ -15,19 +15,6 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-def testattr(**kwargs):
```
Comment (PeterMeisrimelModelon) [inline src/__init__.py:18]: serves no actual function; removing

File: src/solvers/radau5.py [hunk 2]
```diff
@@ -642,11 +642,11 @@ def _step(self, t, y):
     
     def _collocation_pol(self, Z, col_poly, leny):
         
-        col_poly[2*leny:3*leny] = Z[:leny] / self.C[0,0]
```
Comment (PeterMeisrimelModelon) [inline src/solvers/radau5.py:645]: The extra dimension here gave DeprecationWarnings.

File: tests/solvers/test_euler.py [hunk 3]
```diff
@@ -118,17 +117,18 @@ def init_mode(self, solver):
 
 class Test_Explicit_Euler:
     
-    def setUp(self):
+    @classmethod
```
Comment (PeterMeisrimelModelon) [inline tests/solvers/test_euler.py:120]: These are needed for pytest to behave like nose did.

File: tests/solvers/test_sundials.py [hunk 4]
```diff
@@ -181,107 +179,104 @@ def test_event_localizer_external(self):
...
     def test_init(self):
         """
         This tests the functionality of the method __init__.
         """
-        # nose.tools.assert_equal(self.simulator.f, 'Test function')
-        nose.tools.assert_equal(self.simulator.y, 1.0)
-        nose.tools.assert_equal(self.simulator.discr, 'BDF')
-        nose.tools.assert_equal(self.simulator.iter, 'Newton')
-        nose.tools.assert_equal(self.simulator.maxord, 5)
+        # assert self.simulator.f == 'Test function'
```
Comment (modelonrobinandersson) [inline tests/solvers/test_sundials.py:270]: Lets remove this

File: tests/solvers/test_sundials.py [hunk 5]
```diff
@@ -1183,22 +1174,28 @@ def handle_event(self,solver, event_info):
...
-        #nose.tools.assert_raises(Exception, self.simulator._set_algvar, 1.0)
-        nose.tools.assert_raises(Exception, self.simulator._set_algvar, [1,'hej',1])
-        nose.tools.assert_raises(Exception, self.simulator._set_algvar, {'Test':'case'})
-        nose.tools.assert_raises(Exception, self.simulator._set_algvar, [-1,0,1])
-        nose.tools.assert_raises(Exception, self.simulator._set_algvar, [1.0,1.0])
-        nose.tools.assert_raises(Exception, self.simulator._set_algvar, [3.0, 1.0, 1.0])
+        # with pytest.raises(Exception):
+        #     self.simulator._set_algvar(1)
+        # with pytest.raises(Exception):
+        #     self.simulator._set_algvar(1.0)
```
Comment (modelonrobinandersson) [inline tests/solvers/test_sundials.py:1188]: This as well

File: tests/solvers/test_sundials.py [hunk 6]
```diff
@@ -1207,11 +1204,10 @@ def test_algvar(self):
         #self.simulator.algvar = vectorb
         #self.simulator.algvar = vectori
         #self.simulator.algvar = vector
-        #nose.tools.assert_equal(self.simulator.algvar[0], vector[0])
-        #nose.tools.assert_equal(self.simulator.algvar[1], vector[1])
-        #nose.tools.assert_equal(self.simulator.algvar[2], vector[2])
+        #assert self.simulator.algvar[0] == vector[0]
+        #assert self.simulator.algvar[1] == vector[1]
+        #assert self.simulator.algvar[2] == vector[2]
```
Comment (modelonrobinandersson) [inline tests/solvers/test_sundials.py:1209]: And these

File: tests/solvers/test_sundials.py [hunk 7]
```diff
@@ -1368,185 +1362,201 @@ class Prob_CVode(Explicit_Problem):
...
-        nose.tools.assert_equal(self.sim.maxcorS, 10)
-        
-        #nose.tools.assert_raises(Exception, self.sim._set_max_cor_S, 0)
-        nose.tools.assert_raises(Exception, self.sim._set_max_cor_S, 'str')
-        nose.tools.assert_raises(Exception, self.sim._set_max_cor_S, [])
-        #nose.tools.assert_raises(Exception, self.sim._set_max_cor_S, -10)
+        assert self.sim.maxcorS == 10
+        
+        # with pytest.raises(Exception):
+            # self.sim._set_max_cor_S(0)
```
Comment (modelonrobinandersson) [inline tests/solvers/test_sundials.py:1552]: This and a few lines below too

**Review by modelonrobinandersson — APPROVED** [review]
This looks great, just minor comments.
Do you have any average numbers on how much time it takes to run the tests with and without these changes?

**Comment by PeterMeisrimelModelon** [comment]
> This looks great, just minor comments. Do you have any average numbers on how much time it takes to run the tests with and without these changes?

Based on the numbers of the last few GitHub actions runs: ~20 sec with pytest to run the tests, ~36 sec on nose with Python 3.11, however also ~20 sec on current master, without any Python changes, with nose.

File: examples/euler_vanderpol.py [hunk 8]
```diff
@@ -86,7 +85,7 @@ def jac(t,y):
 
     #Basic test
     x1 = y[:,0]
-    nose.tools.assert_less(np.abs(float(x1[-1]) - 1.8601438), 1e-1)
+    assert np.abs(float(x1[-1]) - 1.8601438) < 1e-1
```
Comment (modelonrobinandersson) [inline examples/euler_vanderpol.py:88]: Just a minor comment that we are a bit inconsistent with `np.abs` vs `abs` (scalars in both cases)

**Review by modelonrobinandersson — APPROVED** [review]
Well done!

**Pull request merged**